Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -0,0 +1,347 @@
+                }
+            }
+
+            return true;
+        }
+
+        private static bool TryGetBiggestPersonLoss(decimal[] balances, out decimal biggestLoss)
+        {
+            biggestLoss = decimal.MinValue;
+
+            if (balances.Length < 2)
+            {
+                biggestLoss = 0;
+                return false;
+            }
+
+            for (var i = 0; i < balances.Length; i++)
+            {
+                for (var j = i + 1; j < balances.Length; j++)
+                {
+                    var loss = balances[i] - balances[j];
+
+                    if (loss > biggestLoss)
+                    {
+                        biggestLoss = loss;
+                    }
+                }
+            }
+
+            return true;
+        }
+        
+        private static string GetBiggestLossNames(Person[] people, decimal biggestLoss)
+        {
+            var names = new StringBuilder();
+
+            for (var i = 0; i < people.Length; i++)
+            {
+                var hasBiggestLoss = HasBiggestLoss(people[i].GetBalances(), biggestLoss);
+
+                if (hasBiggestLoss)
+                {
+                    if (names.Length > 0)
+                    {
+                        names.Append(", ");
+                    }
+
+                    names.Append(people[i].GetName());
+                }
+            }
+
+            return names.ToString();
+        }
+
+        private static bool HasBiggestLoss(decimal[] balances, decimal biggestLoss)
+        {
+            for (var i = 0; i < balances.Length; i++)
+            {
+                for (var j = i + 1; j < balances.Length; j++)
+                {
+                    if (IsNumberEqual(balances[i] - balances[j], biggestLoss))
+                    {
+                        return true;
+                    }
+                }
+            }
+
+            return false;
+        }
+
+        private static bool IsNumberEqual(decimal num1, decimal num2)
+        {
+            if (Math.Abs(num1 - num2) < Epsilon)
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        private static string GetLargestCurrentNames(Person[] people, decimal largestCurrentBalance)
+        {
+            var names = new StringBuilder();
+
+            for (var i = 0; i < people.Length; i++)
+            {
+                if (people[i].GetCurrentBalance() == largestCurrentBalance)
+                {
+                    if (names.Length > 0)
+                    {
+                        names.Append(", ");
+                    }
+
+                    names.Append(people[i].GetName());
+                }
+            }
+
+            return names.ToString();
+        }
+
+        private static decimal GetSmallestCurrentBalance(Person[] people)
+        {
+            var smallestBalance = decimal.MaxValue;
+            var name = new StringBuilder();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'name'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5116146)